hamcrest/Hamcrest/Type/IsNumeric.php

```diff
@@ -19,10 +19,36 @@ public function __construct()
 
     public function matches($item)
     {
+        if ($this->is_hexadecimal($item)) {
+
+            return true;
+        }
+
         return is_numeric($item);
     }
 
     /**
+     * This check is necessary because PHP 7 doesn't recognize hexadecimal string as numeric anymore.
```

Comment:
The short/long description describes what method does semantically, but shouldn't expose what it's doing technically. Such descriptions will be outdated pretty fast when code is changed.